speechbrain/dataio/audio_io.py

```diff
@@ -0,0 +1,232 @@
+    
+    Example
+    -------
+    >>> # This is an example, actual file needed to run
+    >>> # audio, sr = load("example.wav")
+    """
+    try:
+        # Read audio file - soundfile returns (frames, channels) or (frames,) for mono
+        audio_np, sample_rate = sf.read(
+            path, 
+            start=frame_offset, 
+            frames=num_frames, 
+            dtype='float32', 
+            always_2d=always_2d
+        )
+        
+        # Convert to torch tensor
+        audio = torch.from_numpy(audio_np)
+        
+        # Convert to requested dtype
+        if dtype != torch.float32:
+            audio = audio.to(dtype)
+        
+        # Handle shape conversion
+        if audio.ndim == 1:
+            # Mono audio as 1D
+            if channels_first:
+                # Need to add channel dimension: (frames,) -> (1, frames)
+                audio = audio.unsqueeze(0)
+        elif audio.ndim == 2:
+            # Multi-channel or explicitly 2D mono
+            if channels_first:
+                # Convert from (frames, channels) to (channels, frames)
+                audio = audio.transpose(0, 1)
+        
+        return audio, int(sample_rate)
+        
+    except Exception as e:
+        raise RuntimeError(f"Failed to load audio from {path}: {e}") from e
+
+
+def save(path, src, sample_rate, subtype="PCM_16"):
+    """Save audio to file using soundfile.
+    
+    Arguments
+    ---------
+    path : str
+        Path where to save the audio file.
+    src : torch.Tensor or numpy.ndarray
+        Audio waveform. Can be:
+        - 1D tensor/array: (frames,) - mono
+        - 2D tensor/array: (channels, frames) or (frames, channels)
+        - 3D tensor/array: (batch, channels, frames) - uses first batch element
+    sample_rate : int
+        Sample rate for the audio file.
+    subtype : str, optional
+        Audio encoding subtype (e.g., 'PCM_16', 'PCM_24', 'PCM_32', 'FLOAT').
+        Default: 'PCM_16'.
+    
+    Returns
+    -------
+    None
+    
+    Example
+    -------
+    >>> # This is an example, actual tensor needed to run
+    >>> # waveform = torch.randn(1, 16000)
+    >>> # save("output.wav", waveform, 16000)
+    """
+    try:
+        # Convert to numpy if needed
+        if isinstance(src, torch.Tensor):
+            audio_np = src.detach().cpu().numpy()
+        else:
+            audio_np = np.asarray(src)
+        
+        # Handle different input shapes
+        if audio_np.ndim == 3:
+            # Batched input: (batch, channels, frames) - take first batch
+            audio_np = audio_np[0]
+        
+        if audio_np.ndim == 2:
+            # Determine if it's (channels, frames) or (frames, channels)
+            # Heuristic: if first dimension is smaller and <= 16, likely channels-first
+            if audio_np.shape[0] <= 16 and audio_np.shape[0] < audio_np.shape[1]:
+                # Convert (channels, frames) to (frames, channels)
+                audio_np = audio_np.T
```

Comment:
Although I expect this inference to be correct nearly all the time, let's make this function match the `torchaudio.save()` signature and add a `channels_first=True` argument